tests/api_test.py

```diff
@@ -222,7 +222,8 @@ def f(x, y, z):
     assert f(one, two, z=three) == 123
     assert len(side) == 2  # but should still cache
 
-    f(one, two, z=np.zeros(3))  # doesn't crash
+    with jax.numpy_dtype_promotion('standard'):
```

Comment:
Is this really supposed to be testing promotion?

Reply:
I don't know if it's "really" supposed to be, but it is. My aim was to modify tests as little as possible. Currently the tests run (and pass) with standard promotion, so after this change ISTM they should still run and pass with standard promotion. What do you think?

Reply:
I can buy a strategy of:
a) first mechanically enable dtype promotion everywhere it is needed,
b) then go and clean up its uses
if you want to do it that way.

However I don't want our tests to be littered with overrides on the promotion behavior just because we happened (casually) to use promotion in the past...

For example, in this particular case, I think we were just lazy about the type of `z`.

Reply:
Wait, but here in the next line the comment calls out the fact that a different type is used. So either (1) we keep it as-is with the dtype promotion standard, or (2) we change the test so that it's not checking for a change in `len(side)`.

I think (2) is a bad idea because it removes test coverage. I'm going to stick with (1), as it currently is.